File: Data Structures & Algorithms/balanced-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private boolean balanced=true;
    public boolean isBalanced(TreeNode root) {
        if(root==null) return true;
        height(root);
        return balanced;
    }
    public int height(TreeNode root){
        if(root==null||!balanced) return 0;
        int lefth=height(root.left);
        int righth=height(root.right);
        if(Math.abs(lefth-righth)>1) balanced =false;

        return Math.max(lefth,righth)+1;
    }
}

File: Data Structures & Algorithms/pow-x-n/submission-1.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if(x==0) return 0.0;
        if(n==0) return 1.0;
        if(n==1) return x;
       
       double a=1.0;
       long long exp=abs((long long)n);
       while(exp>0){
        if(exp%2==1){
            a=a*x;
        }
        x=x*x;
        exp/=2;
       }
       return n>0? a : 1/a;
    }
};
